mongodbreactivestreams/src/main/java/site/ycsb/db/OptionsSupport.java

```diff
@@ -0,0 +1,145 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package site.ycsb.db;
+
+import java.util.Properties;
+
+/**
+ * OptionsSupport provides methods for handling legacy options.
+ *
+ * @author rjm
+ */
+public final class OptionsSupport {
+
+  /** Value for an unavailable property. */
+  private static final String UNAVAILABLE = "n/a";
+
+  /**
+   * Updates the URL with the appropriate attributes if legacy properties are
+   * set and the URL does not have the property already set.
+   *
+   * @param url
+   *          The URL to update.
+   * @param props
+   *          The legacy properties.
+   * @return The updated URL.
+   */
+  public static String updateUrl(String url, Properties props) {
+    String result = url;
+
+    // max connections.
+    final String maxConnections =
+        props.getProperty("mongodb.maxconnections", UNAVAILABLE).toLowerCase();
+    if (!UNAVAILABLE.equals(maxConnections)) {
+      result = addUrlOption(result, "maxPoolSize", maxConnections);
+    }
+
+    // Blocked thread multiplier.
+    final String threadsAllowedToBlockForConnectionMultiplier =
+        props
+            .getProperty(
+                "mongodb.threadsAllowedToBlockForConnectionMultiplier",
+                UNAVAILABLE).toLowerCase();
+    if (!UNAVAILABLE.equals(threadsAllowedToBlockForConnectionMultiplier)) {
+      result =
+          addUrlOption(result, "waitQueueMultiple",
+              threadsAllowedToBlockForConnectionMultiplier);
+    }
+
+    // write concern
+    String writeConcernType =
+        props.getProperty("mongodb.writeConcern", UNAVAILABLE).toLowerCase();
+    if (!UNAVAILABLE.equals(writeConcernType)) {
+      if ("errors_ignored".equals(writeConcernType)) {
+        result = addUrlOption(result, "w", "0");
+      } else if ("unacknowledged".equals(writeConcernType)) {
+        result = addUrlOption(result, "w", "0");
+      } else if ("acknowledged".equals(writeConcernType)) {
+        result = addUrlOption(result, "w", "1");
+      } else if ("journaled".equals(writeConcernType)) {
+        result = addUrlOption(result, "journal", "true"); // this is the
+        // documented option
+        // name
+        result = addUrlOption(result, "j", "true"); // but keep this until
+        // MongoDB Java driver
+        // supports "journal" option
+      } else if ("replica_acknowledged".equals(writeConcernType)) {
+        result = addUrlOption(result, "w", "2");
+      } else if ("majority".equals(writeConcernType)) {
+        result = addUrlOption(result, "w", "majority");
+      } else {
+        System.err.println("WARNING: Invalid writeConcern: '"
```

Comment:
Prefer to use a logger over s.e.p

Reply:
Same comment applies throughout PR :-).

Reply:
Thanks for the feedback. We have a work item for upgrading to log4j2 and adding micrometer as we did for azurecosmos binding.
#19 
#20

Reply:
https://github.com/Azure/YCSB/issues/19 completed by Ravi , moved everything to log4j